src/derivers/bip39.ts

```diff
@@ -1,17 +1,59 @@
-import { mnemonicToEntropy, mnemonicToSeed } from '@metamask/scure-bip39';
+import { mnemonicToEntropy } from '@metamask/scure-bip39';
 import { wordlist as englishWordlist } from '@metamask/scure-bip39/dist/wordlists/english';
-import { assert } from '@metamask/utils';
-import { hmac } from '@noble/hashes/hmac';
-import { pbkdf2 } from '@noble/hashes/pbkdf2';
-import { sha512 } from '@noble/hashes/sha512';
+import { assert, stringToBytes } from '@metamask/utils';
 
 import type { DeriveChildKeyArgs } from '.';
 import type { BIP39StringNode } from '../constants';
 import { BYTES_KEY_LENGTH } from '../constants';
+import { hmacSha512, pbkdf2Sha512 } from '../cryptography';
 import type { Curve } from '../curves';
 import { SLIP10Node } from '../SLIP10Node';
 import { getFingerprint } from '../utils';
 
+/**
+ * Encode a BIP-39 mnemonic phrase to a `Uint8Array` for use in seed generation.
+ * If the mnemonic is already a `Uint8Array`, it is assumed to contain the
+ * indices of the words in the wordlist.
+ *
+ * @param mnemonic - The mnemonic phrase to encode.
+ * @param wordlist - The wordlist to use.
+ * @returns The encoded mnemonic phrase.
+ */
+function encodeMnemonicPhrase(
+  mnemonic: string | Uint8Array,
+  wordlist: string[],
+) {
+  if (typeof mnemonic === 'string') {
```

Comment:
In the scure implementation, this will throw if the mnemonic isn't of the right size